.delivery/build-cookbook/recipes/deploy.rb

```diff
@@ -4,3 +4,57 @@
 #
 # Copyright (c) 2015 The Authors, All Rights Reserved.
 include_recipe 'delivery-truck::deploy'
+
+#
+# We publish to supermarket.chef.io when a change hits delivered
+#
+# TODO: The following code needs to be incorporated into `delivery-truck`
+#       in some manner.
+#
+if delivery_environment == 'delivered'
+  require 'uri'
+
+  supermarket_creds = encrypted_data_bag_item_for_environment('creds', 'supermarket')
```

Comment:
Just to be clear, **delivery-truck** does not depends on **chef-sugar**. That is why I suggested to add it to your `metadata.rb`